java-connectors/kafka-connect-gcp-pubsub/src/main/java/io/lenses/streamreactor/connect/gcp/pubsub/source/GCPPubSubSourceConnector.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.pubsub.source;
+
+import static io.lenses.streamreactor.common.util.AsciiArtPrinter.printAsciiHeader;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.source.SourceConnector;
+
+import io.lenses.streamreactor.common.util.JarManifest;
+import io.lenses.streamreactor.common.util.TasksSplitter;
+import io.lenses.streamreactor.connect.gcp.pubsub.source.configdef.PubSubConfigDef;
+
+/**
+ * GCPPubSubSourceConnector is a source connector for Google Cloud Pub/Sub.
+ * It is responsible for starting the connector, creating tasks, and stopping the connector.
+ */
+public class GCPPubSubSourceConnector extends SourceConnector {
+
+  private Map<String, String> props;
+
+  private final JarManifest jarManifest =
+      new JarManifest(getClass().getProtectionDomain().getCodeSource().getLocation());
+
+  @Override
+  public void start(Map<String, String> props) {
+    this.props = props;
```

Comment:
Do we need to validate the properties here? Imagine the KCQL is wrong, won't it start the tasks and fail? so doing it here we could fail fast?